app/src/main/kotlin/com/wire/android/ui/home/conversations/MessageItemComponents.kt

```diff
@@ -74,6 +78,123 @@ internal fun MessageSendFailureWarning(
     }
 }
 
+@Composable
+internal fun MessageSentPartialDeliveryFailures(partialDeliveryFailureContent: DeliveryStatusContent.PartialDelivery) {
+    val resources = LocalContext.current.resources
+    CompositionLocalProvider(
+        LocalTextStyle provides MaterialTheme.typography.labelSmall.copy(color = MaterialTheme.wireColorScheme.error)
+    ) {
+        if (partialDeliveryFailureContent.isSingleUserFailure) {
+            SingleUserDeliveryFailure(partialDeliveryFailureContent, resources)
+        } else {
+            MultiUserDeliveryFailure(partialDeliveryFailureContent, resources)
+        }
+    }
+}
+
+@Composable
+private fun MultiUserDeliveryFailure(
+    partialDeliveryFailureContent: DeliveryStatusContent.PartialDelivery,
+    resources: Resources
+) {
+    var expanded: Boolean by remember { mutableStateOf(false) }
+    Column {
+        Text(
+            text = stringResource(
+                id = R.string.label_message_partial_delivery_participants_count,
+                partialDeliveryFailureContent.totalUsersWithFailures
+            ),
+            textAlign = TextAlign.Start
+        )
+        VerticalSpace.x4()
+        if (expanded) {
+            if (partialDeliveryFailureContent.filteredRecipientsFailure.isNotEmpty()) {
+                Text(
+                    text = stringResource(
+                        id = R.string.label_message_partial_delivery_participants_deliver_later,
+                        partialDeliveryFailureContent.filteredRecipientsFailure
+                            .filter {
+                                !it.asString(resources).contentEquals(resources.getString(R.string.username_unavailable_label))
```

Comment:
Do you need to pass resources when you are in composable function?

Reply:
Thanks! :D I'm taking note of this, and I will open a small pr with these improvements to avoid blocking builds on develop